orders/views.py

```diff
@@ -1,3 +1,77 @@
-from django.shortcuts import render
+import json
+from json.decoder import JSONDecodeError
 
-# Create your views here.
+from django.views import View
+from django.http  import JsonResponse
+from django.db    import transaction
+
+from .models         import User, Order, Cart
+from products.models import Product
+
+class CartView(View):
+    @transaction.atomic
+    def post(self, request):
+        STATUS_IN_CART = 1
+
+        user       = request.user
+        data       = json.loads(request.body)
+        product_id = data['id']
+        count      = data['count']
+
+        save_point = transaction.savepoint()
```

Comment:
order와 cart 정보에 대해서만 transaction을 사용할 경우 save_point를 따로 지정할 필요가 없어 보입니다.
가독성 측면에서도 그렇고, 코드 낭비입니다.
어짜피 order 생성 이전으로 save_point를 남기는 거라면 `@transaction.atomic`만 사용하는 것과 다른게 없습니다.

```suggestion
        product_id = data['id']
        count      = data['count']

        save_point = transaction.savepoint()
```

Reply:
ㅠ.ㅠ 데코레이터만 사용했을 때는 트랜잭션이 작동하지 않아서..
savepoint를 사용해봤더니 트랜잭션이 정상적으로 되더라구요...ㅠㅠ
수정할게요!!